Add multiple export formats inference support in `Live Inference` solution

RizwanMunawar opened the pull request:
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improves Streamlit live inference by simplifying custom model selection and expanding supported model formats, plus a small doc cleanup. 🚀

### 📊 Key Changes
- Streamlit UI now accepts full model paths directly (no auto-stripping of “.pt”). 🔧
- Automatically detects and loads multiple model formats: .pt, .onnx, .torchscript, .mlpackage, .engine, and OpenVINO exports. 📦
- Defaults to appending “.pt” only when the user selects a preset model name. ✅
- Removed an outdated docs snippet that suggested passing a model without the file extension. 🧹

### 🎯 Purpose & Impact
- More flexible model loading: users can plug in any supported format without renaming or extension hacks. 🎯
- Fewer loading errors and confusion around file extensions in Streamlit apps. 🧠
- Cleaner docs aligned with current behavior, reducing setup friction. 📚
- Better compatibility for deployment scenarios using non-.pt exports (e.g., ONNX, TensorRT, OpenVINO). ⚙️

RizwanMunawar committed:
Add support for .pt, .onnx, .torchscript, .mlpackage exported models inference

UltralyticsAssistant commented:
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To help us review and integrate your contribution efficiently, please take a moment to review this checklist:

- ✅ **Define a Purpose**: Make sure your PR description clearly outlines the purpose and impact of your proposed changes, and link any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Please ensure your commit messages remain clear and concise.
- ✅ **Synchronize with Source**: Double-check that your branch is up to date with the `ultralytics/ultralytics` `main` branch. If it’s out of sync, please update it by clicking 'Update branch' or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Please verify that all [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks have passed. If not, kindly address any issues that arise.
- ✅ **Update Documentation**: If your changes introduce new features or modify existing functionality, update the relevant [documentation](https://docs.ultralytics.com/) as needed.
- ✅ **Add Tests**: Include or update tests to cover your enhancements, and confirm that all tests pass.
- ✅ **Sign the CLA**: If this is your first time contributing, please sign our [Contributor License Agreement (CLA)](https://docs.ultralytics.com/help/CLA/) by commenting "I have read the CLA Document and I sign the CLA" in this PR.
- ✅ **Minimize Changes**: Limit your PR to only the essential modifications necessary for this enhancement. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For a detailed overview, refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have any questions, feel free to leave a comment below.

⚡ This is an automated response. An Ultralytics engineer will review your PR and assist you further soon. Thank you for helping improve Ultralytics! 🎉

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

RizwanMunawar committed:
update docs

RizwanMunawar commented:
@Y-T-G @Laughing-q Hi guys, here, I've added support for inference with different export formats based on user request #21517.

### Step#01: Export Ultralytics YOLO11 model to ONNX

```bash
yolo export format=onnx
```

**Step#02: Run Inference using Streamlit app with exported model*

```python
from ultralytics import solutions

inf = solutions.Inference(
    model="yolo11n.onnx",  # you can use any model that Ultralytics support, i.e. YOLO11, or custom trained model
)

inf.inference()
```

Please let me know what you think. Thanks

Y-T-G reviewed (commented):
Hmm. Is there an issue if we just allow any model extension? And only manually add `.pt` if the `selected_model` doesn't have any extension?

RizwanMunawar commented:
@Y-T-G, yes, that's a good point. I was also thinking about that. But I realized other formats are not widely used, and are mostly for specific devices. I mean, adding all formats isn't a problem, we can add these, but it's just I don't think these will be used alot.

RizwanMunawar committed:
Add TensorRT `.engine` export format inference support

@Y-T-G as discussed :)

Signed-off-by: Muhammad Rizwan Munawar <muhammadrizwanmunawar123@gmail.com>

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

UltralyticsAssistant committed:
Merge branch 'main' into mm-streamlit

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

Laughing-q committed:
Merge branch 'main' into mm-streamlit

RizwanMunawar committed:
Merge 7f24a0a87cd2917b3cfa01c57210a232cb3ee99a into 809157e70fea05bbd6f42625d373bda29a710424

UltralyticsAssistant committed:
Auto-update Ultralytics Docs Reference by https://ultralytics.com/actions

Laughing-q commented:
@RizwanMunawar and also why the model name are different in the interface?
```bash
yolo solutions inference
```
<img width="465" height="728" alt="pic-250808-1159-04" src="https://github.com/user-attachments/assets/6845c10a-e39d-4ee6-97e1-7e00eac11029" />
all the other are capitalized and the top one is not

RizwanMunawar committed:
remove duplication python code example

RizwanMunawar commented:
@Laughing-q, you're right. The second example is indeed redundant. It was originally introduced in this [PR](https://github.com/ultralytics/ultralytics/pull/14563/files), which I hadn’t reviewed at that time, so I hadn’t removed or modified it earlier.

But yes, it’s a clear duplication, and I’ve now removed it based on your suggestion here: [8598190](https://github.com/ultralytics/ultralytics/pull/21553/commits/85981909f7a9457122da7572071931d7d58a1fcf)

**CLI commands**:

```bash
 # Run the app without the custom model and use the default pre-trained models that we have in the assets.
yolo solutions inference  

# Allow users to run their custom models with a streamlit app.
yolo solutions inference model="path/to/model.pt" 
```

The difference in the first model name in the dropdown is becuase it's the default argument [here](https://github.com/ultralytics/ultralytics/blob/809157e70fea05bbd6f42625d373bda29a710424/ultralytics/cfg/__init__.py#L708), which I adjusted to align with CLI usage. All other models are different because they’re sourced directly from the [assets](https://github.com/ultralytics/ultralytics/blob/809157e70fea05bbd6f42625d373bda29a710424/ultralytics/solutions/streamlit_inference.py#L158).

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

glenn-jocher committed:
Merge branch 'main' into mm-streamlit

RizwanMunawar committed:
Merge branch 'main' into mm-streamlit

Laughing-q committed:
Merge branch 'main' into mm-streamlit
